src/validation/validateSubmissions.ts

```diff
@@ -0,0 +1,256 @@
+import * as gqlTypes from '../typeDefs/gqlTypes.js'
+import { ErrorCode, Result } from '../result.js'
+import { hasScriptTags, isInvalidName } from '../../utils/stringUtils.js'
+
+function validateSubmissionOrderBy(
+    filters: gqlTypes.SubmissionSearchFilters,
+    validateSearchResults: Result<unknown>
+    ): void {
+
+    if (!filters.orderBy || filters.orderBy.length === 0) {
+        return
+    }
+    
+    const validOrderByFields = [
+        'googleMapsUrl',
+        'healthcareProfessionalName',
+        'isUnderReview',
+        'isApproved',
+        'isRejected',
+        'createdDate',
+        'updatedDate',
+    ]
+
+    if (filters.orderBy?.some(order => order.fieldToOrder === 'id')) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_OPTION_INVALID,
+            httpStatus: 400
+        })
+    }
+
+    const invalidOrderByOption = filters.orderBy?.find(order => 
+        !validOrderByFields.includes(order.fieldToOrder))
+
+    if (invalidOrderByOption) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_OPTION_NOT_FOUND,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.orderBy && filters.orderBy.length > 2) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_OPTION_MAX_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.orderBy && filters.orderBy.length === 2 && 
```

Comment:
no need to check `orderby` for null after the first check since you return early 👍 

you can remove any of the future checks